src/emitter/emitter.ts

```diff
@@ -35,6 +49,39 @@ export class EventEmitter<E extends object> implements IEventEmitter<E> {
   public dispose() {
     this._emitter.removeAllListeners();
   }
+
+  public createEmitGroup(): {
+    queueCount: number;
+    emit: EventEmitter<E>['emit'];
+    flush: () => void;
+    reset: () => void;
+  } {
+    const notifier = this;
+
+    let eventQueue: Array<[unknown, Array<unknown>]> = [];
+
+    return {
+      get queueCount() {
+        return eventQueue.length;
+      },
+
+      emit(event: any, ...args: any) {
```

Comment:
I believe we can narrow it down to string and numbers.
```suggestion
      emit(event: string | number, ...args: any) {
```

Reply:
Moreover, I can't help but feel we may want to name it `queue` or `schedule` rather than `emit`? Fine with emit though, just sayin'.

Reply:
The intent here is that downstream consumers (like patcher in graphite) don't need to change - they're calling `emit` and the callee of patcher that passes in the notifier can decide wether to accumulate in an emitGroup, or pass notifier that emits on every call.